blocklogger_test.go

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/decred/dcrd/dcrutil/v3"
+	"github.com/decred/dcrd/wire"
+)
+
+func TestLogBlockHeight(t *testing.T) {
+
```

Comment:
There might be some other cases left in the code that creates the type separate, but we've been working towards making the tests more consistent in that the types are just defined inline and the tests definitions are "compactified" to reduce pushing the important stuff so far to the right.  Along those lines, the way this is all either already done throughout, or is being converted to over time, is as follows:

```Go
func TestFoo(t *testing.T) {
	tests := []struct {
		name                       string
		reset                      bool
		inputBlock                 *dcrutil.Block
		inputSyncHeight            int64
		inputLastLogTime           time.Time
		wantReceivedLogBlocks      int64
		wantReceivedLogTx          int64
		wantReceivedLogVotes       int64
		wantReceivedLogRevocations int64
		wantReceivedLogTickets     int64
	}{{
		name:       "descriptive name of test condition here",
		inputBlock: dcrutil.NewBlock(&TestBlock100000),
		// ...
	}, {
		name:       "descriptive name of next test condition here",
		inputBlock: dcrutil.NewBlock(&TestBlock100001),
		// ...
	}, {
		// ...
	}, {
		name:       "descriptive name of test condition here",
		reset: true,
		inputBlock: dcrutil.NewBlock(&TestBlock100000),
		// ...
	}}

	progressLogger := newBlockProgressLogger("Written", bmgrLog)
	for _, test := range tests {
		if test.reset {
			progressLogger = newBlockProgressLogger("Written", bmgrLog)
		}

		progressLogger.SetLastLogTime(testCase.inputLastLogTime)
		progressLogger.logBlockHeight(testCase.inputBlock, testCase.inputSyncHeight)
		// ...

		if !reflect.DeepEqual(progressLogger, wantBlockProgressLogger) {
			t.Errorf("%s:\nwant: %+v\ngot: %+v\n", test.name,
				wantBlockProgressLogger, progressLogger)
		}
	}
}
```

Reply:
Thanks for the feedback. Makes sense to me and made adjustments based on both of the points that you brought up.